vcd/resource_vcd_nsxt_edgegateway_dns.go

```diff
@@ -0,0 +1,373 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdNsxtEdgeGatewayDns() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   resourceVcdNsxtEdgegatewayDnsRead,
+		CreateContext: resourceVcdNsxtEdgegatewayDnsCreate,
+		UpdateContext: resourceVcdNsxtEdgegatewayDnsUpdate,
+		DeleteContext: resourceVcdNsxtEdgegatewayDnsDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNsxtEdgegatewayDnsImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"edge_gateway_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway ID for DNS configuration",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Status of the DNS Forwarder. Defaults to `true`",
+			},
+			"listener_ip": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				Description: "IP on which the DNS forwarder listens." +
+					"Can be modified only if the Edge Gateway has a dedicated external network.",
+				ValidateFunc: validation.IsIPAddress,
+			},
+			"snat_rule_enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "The value is `true` if a SNAT rule exists for the DNS forwarder.",
+			},
+			"snat_rule_ip_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				Description: "The external IP address of the SNAT rule. " +
+					"Can be modified only if the Edge Gateway's external network is using IP spaces. (VCD 10.5.0+)",
+			},
+			"default_forwarder_zone": {
+				Type:        schema.TypeList,
+				Required:    true,
+				MaxItems:    1,
+				Description: "The default forwarder zone.",
+				Elem:        defaultForwarderZone,
+			},
+			"conditional_forwarder_zone": {
```

Comment:
Where does the limit come from? Is it an API limitation?

Reply:
It isn't mentioned in the API documentation, but in the UI there's this box:
![image](https://github.com/vmware/terraform-provider-vcd/assets/16798732/913dd5c9-6950-4909-9b3d-1fc7f8138522)